src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -4442,13 +4444,27 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
 
             head.m_v = ASRUtils::EXPR(explicit_iter_var);
         } else {
-            body.reserve(al, x.n_body);
+            body.reserve(al, x.n_body + 2);
             head.m_v = target;
         }
 
         SymbolTable *parent_scope = current_scope;
         current_scope = al.make_new<SymbolTable>(parent_scope);
+        std::string name = "j";
+        ASR::expr_t *j = ASR::down_cast<ASR::expr_t>(ASR::make_Var_t(
+            al, x.base.base.loc, ASR::down_cast<ASR::symbol_t>(ASR::make_Variable_t(
+                al, x.base.base.loc, current_scope, s2c(al, name), nullptr, 0, ASR::intentType::Local,
+                nullptr, nullptr, ASR::storage_typeType::Default,
+                ASRUtils::TYPE(ASR::make_Integer_t(al, x.base.base.loc, 4, nullptr, 0)),
+                ASR::abiType::Source, ASR::accessType::Public,
+                ASR::presenceType::Required, false))));
+        assign_to_j = ASRUtils::STMT(ASR::make_Assignment_t(al, x.base.base.loc, j, target, nullptr));
+        i_update = ASRUtils::STMT(ASR::make_Assignment_t(al, x.base.base.loc, target,
+                    ASRUtils::EXPR(ASR::make_IntegerBinOp_t(al, x.base.base.loc, j,
+                    ASR::binopType::Add, constant_one, a_type, nullptr)), nullptr));
```

Comment:
Can you check where `i` value is incremented, and replace this with that or vice-versa?
 I think this will make `i` to be incremented twice.

Reply:
Hmm correct , maybe we could just assign `j` back to `i` after the for loop body ?

Reply:
I think so

Reply:
Thanks for the prompt , here my latest commit fixes all sorts of local increments . Few cases are shown below
```
(lf) anutosh491@spbhat68:~/lpython/lpython$ cat examples/expr2.py 
def main():
    i: i32
    for i in range(5):
        print(i)
        i=i+1

main()


(lf) anutosh491@spbhat68:~/lpython/lpython$ ./src/bin/lpython examples/expr2.py 
0
1
2
3
4
```
```
(lf) anutosh491@spbhat68:~/lpython/lpython$ cat examples/expr2.py 
def main():
    i: i32
    for i in range(5):
        i = i*10
        print(i)

main()

(lf) anutosh491@spbhat68:~/lpython/lpython$ ./src/bin/lpython examples/expr2.py 
0
10
20
30
40
```
```
(lf) anutosh491@spbhat68:~/lpython/lpython$ cat examples/expr2.py 
def main():
    i: i32
    lst: list[i32] = [1, 2, 3]
    for i in lst:
        print(i)
        i = i*10

main()

(lf) anutosh491@spbhat68:~/lpython/lpython$ ./src/bin/lpython examples/expr2.py 
1
2
3
```

Reply:
But as we can see tests fail , I guess this is because of reference files being updated . Once you go through the code and give me a green flag , I'll update the references too . cc @Thirumalai-Shaktivel  (Is this all ? maybe this would need an integration test or something added in tests.toml , you could let me know ...)